pkg/probe/turbo_probe.go

```diff
@@ -9,36 +9,63 @@ import (
 )
 
 // Turbo Probe Abstraction
-// Consists of clients that handle probe registration metadata, discovery and action execution for different probe targets
+// Consists of clients that handle probe registration metadata,
+// and the discovery and action execution for different probe targets
 type TurboProbe struct {
-	ProbeType          string
-	ProbeCategory      string
-	RegistrationClient TurboRegistrationClient
-	DiscoveryClientMap map[string]TurboDiscoveryClient
+	ProbeConfiguration *ProbeConfig
+	RegistrationClient *ProbeRegistrationAgent
+	DiscoveryClientMap map[string]*TargetDiscoveryAgent
 
 	ActionClient TurboActionExecutorClient
 }
 
+type ProbeRegistrationAgent struct {
```

Comment:
These interfaces are all for `registrationClient`, shall we merge these **Five** interfaces into  **one** interface? 

So that, when we implement a `registrionClient`, we can know which functions are needed to be implemented in one shot?